### PR TITLE
Named branch configuration

### LINKED ISSUE #995: Alias branch types in config
Lets be honest. Regexes as keys kinda sucks...

```
branches:
  (pull|pull\-requests|pr)[/-]:
    tag-name-pattern: '[/-](?<number>\d+)[-/]'
```

Lets fix this in v4 and change to something like:

```
branches:
  pullrequest:
    match: '(pull|pull\-requests|pr)[/-]'
    tag-name-pattern: '[/-](?<number>\d+)[-/]'
```

### PR DESCRIPTION
**Pull request by JakeGinnivan**
// @asbjornu @gep13 

No implementation yet, but happy with the docs/test changes before I implement

Fixes #995

### CODE REVIEW

**Comment by JakeGinnivan** [comment]
@GitTools/developers would love a review on this, it's a decent change. Anything you can see which can be improved?

File: src/GitVersionCore.Tests/ConfigProviderTests.cs [hunk 1]
```diff
@@ -29,46 +29,26 @@ public void Setup()
...
-        const string text = @"
-assemblyVersioningScheme: MajorMinor
-develop-branch-tag: alpha
-release-branch-tag: rc
+       tag: rc
 ";
         SetupConfigFileContent(text);
         var error = Should.Throw<OldConfigurationException>(() => ConfigurationProvider.Provide(repoPath, fileSystem));
         error.Message.ShouldContainWithoutWhitespace(@"GitVersion configuration file contains old configuration, please fix the following errors:
+GitVersion branch configs no longer are keyed by regexes, update:
```
Comment (JakeGinnivan) [inline src/GitVersionCore.Tests/ConfigProviderTests.cs]: I put a bit of effort into the error messages to give a good upgrade experience.

**Comment by pascalberger** [comment]
What about https://github.com/GitTools/GitVersion/blob/master/BREAKING%20CHANGES.md? IMHO it would be nice to have this updated too with all the breaking changes like this and others introduced in 4.0

**Comment by JakeGinnivan** [comment]
@pascalberger nice idea, done.

**Comment by asbjornu** [comment]
@JakeGinnivan Excellent change! :ok_hand:

**Pull request merged**